go/rpcclientlib/encrypted_client.go

```diff
@@ -49,14 +52,28 @@ func NewEncRPCClient(client Client, viewingKey *ViewingKey) (*EncRPCClient, erro
 	return encClient, nil
 }
 
-// Call handles JSON rpc requests - if the method is sensitive it will encrypt the args before sending the request and
-// then decrypts the response before returning.
-// The result must be a pointer so that package json can unmarshal into it. You can also pass nil, in which case the result is ignored.
+// Call handles JSON rpc requests without a context - see callInternal for details
 func (c *EncRPCClient) Call(result interface{}, method string, args ...interface{}) error {
+	return c.callInternal(c.obscuroClient.Call, result, method, args...)
+}
+
+// CallContext will return immediately if ctx is cancelled - see callInternal for details
+func (c *EncRPCClient) CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error {
+	callExec := func(result interface{}, method string, args ...interface{}) error {
+		return c.obscuroClient.CallContext(ctx, result, method, args...)
+	}
+	return c.callInternal(callExec, result, method, args...)
+}
+
+// callInternal is the main logic to execute JSON-RPC requests:
+// - if the method is sensitive it will encrypt the args before sending the request and then decrypts the response before returning
+// - result must be a pointer so that package json can unmarshal into it. You can also pass nil, in which case the result is ignored.
+// - callExec handles the delegated call, allows EncClient to use the same code for calling with or without a context
+func (c *EncRPCClient) callInternal(callExec callExecutor, result interface{}, method string, args ...interface{}) error {
 	assertResultIsPointer(result)
 	if !isSensitive(method) {
 		// for non-sensitive methods or when viewing keys are disabled we just delegate directly to the geth RPC client
-		return c.obscuroClient.Call(result, method, args...)
+		return callExec(result, method, args...)
```

Comment:
> “Your scientists were so preoccupied with whether or not they could, they didn't stop to think if they should.”

I think this is pretty wacky tbh. Can't you just pass a context object into `callInternal`, then have an if-statement where if the context is nil, you use `Call`, and otherwise you use `CallContext`? Seems like it would be much easier to follow.